File: designParkingSystem.py
class ParkingSystem:
    def __init__(self, big: int, medium: int, small: int):
        self.bigSpot = big
        self.mediumSpot = medium
        self.smallSpot = small
        
    def addCar(self, carType: int) -> bool:
        if (carType == 1):
            if (self.bigSpot == 0):
                result = False
            else:
                self.bigSpot = self.bigSpot - 1
                result = True
                
        elif (carType == 2):
            if (self.mediumSpot == 0):
                result = False
            else:
                self.mediumSpot = self.mediumSpot - 1
                result = True
                
        elif (carType == 3):
            if (self.smallSpot == 0):
                result = False
            else:
                self.smallSpot = self.smallSpot - 1
                result = True
        
        return result

myObj = ParkingSystem(1,1,0)
print(myObj.addCar(1))
print(myObj.addCar(2))
print(myObj.addCar(3))
print(myObj.addCar(1))

File: designParkingSystem.js
/**
 * @param {number} big
 * @param {number} medium
 * @param {number} small
 */
var bigSlot, mediumSlot, smallSlot;
var result;

var ParkingSystem = function(big, medium, small) 
{
    bigSlot = big;
    mediumSlot = medium;
    smallSlot = small;
};

/** 
 * @param {number} carType
 * @return {boolean}
 */
ParkingSystem.prototype.addCar = function(carType) {
      if (carType == 1)
        {
            if (bigSlot == 0)
            {
                result = false;
            }
            
            else
            {
                bigSlot--;
                result = true;
            }
        }
        
        else if (carType == 2)
        {
            if (mediumSlot == 0)
            {
                result = false;
            }
            
            else
            {
                mediumSlot--;
                result = true;
            }
        }
        
        else if (carType == 3)
        {
            if (smallSlot == 0)
            {
                result = false;
            }
            
            else
            {
                smallSlot--;
                result = true;
            }
        }
        
        return result;
};

var myObj = new ParkingSystem(1,1,0);

console.log(myObj.addCar(1));
console.log(myObj.addCar(2));
console.log(myObj.addCar(3));
console.log(myObj.addCar(1));



/** 
 * Your ParkingSystem object will be instantiated and called as such:
 * var obj = new ParkingSystem(big, medium, small)
 * var param_1 = obj.addCar(carType)
 */
